clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -624,10 +624,8 @@ public boolean commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets,
 
     private void maybeAutoCommitOffsetsAsync(long now) {
         if (autoCommitEnabled) {
-            if (coordinatorUnknown()) {
-                this.nextAutoCommitDeadline = now + retryBackoffMs;
-            } else if (now >= nextAutoCommitDeadline) {
-                this.nextAutoCommitDeadline = now + autoCommitIntervalMs;
+            if (now >= nextAutoCommitDeadline) {
+                this.nextAutoCommitDeadline = now + (coordinatorUnknown() ? retryBackoffMs : autoCommitIntervalMs);
```

Comment:
I think a better approach might be to set `nextAutoCommitDeadline` in the offset commit callback in `doAutoCommitOffsetsAsync`. If there was an error, we can set it based on the retry backoff; otherwise we can use the auto commit interval.